metaseq/data/filter_dataset.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import numpy as np
+import pandas as pd
+import json
+
+from metaseq.file_io import PathManager
+
+from metaseq.data import data_utils
+from . import BaseWrapperDataset
+
+class FilterDataset(BaseWrapperDataset):
+    """Filters dataset by excluding examples that are easy/hard to learn.
+    
+    Hard/easy to learn are defined by a metric between 0.0 and 1.0, with
+    0.0 meaning easy to learn, and 1.0 meaning hard to learn.
+
+    During initialization, this class expects:
+    - `frac_data`: how much of the original dataset we want to keep 
+    during training. It calculates the closest `frac_data` we can keep
+    (keep the hard examples, throw away easy examples, 
+    as per https://arxiv.org/abs/2206.14486.
+    - `metric_file`: path to jsonl file, where each line should be
```

Comment:
The reason I chose the specified format for the metrics file: in the future, if we want to compute metrics that are done offline (do not involve running a metaseq model), we can still use that metric to filter data points at train time